### ISSUE TITLE
UP036 multiple conditions false negative

### ISSUE THREAD

**Issue by gamer191** [astral-sh/ruff#12093]
UP036 doesn't work if there are multiple conditions. For a file Test2.txt with the text:
```
import sys
if True and sys.version_info < (3, 5):
    print("35")
```
running `ruff check Test2.txt --extend-select UP --config "target-version = 'py38'" --isolated` will result in `All checks passed!`
```
>ruff --version
ruff 0.5.0

>cat Test2.txt
import sys
if True and sys.version_info < (3, 5):
    print("35")


>ruff check Test2.txt --extend-select UP --config "target-version = 'py38'" --isolated
All checks passed!

```

**Comment by dhruvmanila** [comment]
It seems reasonable to check for conditions although that means that we'd need to get a range of version if there are multiple `sys.version_info` checks like:
```py
if sys.version_info > (3, 5) and sys.version_info < (3, 10): ...
```
Here, we might want to raise `UP036` for `(3, 5)` and could restrict it to the `target-version`. I'm curious to hear others thoughts on this.

**Comment by gamer191** [comment]
> if sys.version_info > (3, 5) and sys.version_info < (3, 10): ...

I think both should be considered individually. `sys.version_info > (3, 5)` is not allowed unless the target version is set to 3.5 (even then, it's not currently allowed due to #8832), so that should raise an error. `sys.version_info < (3, 10)` should raise another error if the target version is 3.10 or above (I don't know if raising multiple identical errors for one line is possible)

The same should obviously apply for:
```
if sys.version_info < (3, 5) or sys.version_info > (3, 10):
```

Real world example (which I've reported to the yt-dlp devs): https://github.com/yt-dlp/yt-dlp/blob/4862a29854d4044120e3f97b52199711ad04bee1/yt_dlp/compat/__init__.py#L39

**Comment by Avasam** [comment]
I agree with @gamer191 here. Each condition should be considered individually. It's fine if you stop attempting to provide an autofix as soon as there's multiple conditions. But I would expect Ruff to flag `UP036` here.
